### PR TITLE
Swik 1412 merge from template

### PR DESCRIPTION
**Pull request by TBoonX**
I merged the new version of the template.
I had to fix alot of linter errors

### CODE REVIEW

File: application/db_migration_usernames.js [hunk 1]
```diff
@@ -19,34 +22,34 @@ return helper.connectToDatabase()
               return;
 
             this.email = this.email.toLowerCase();
-            emit(this.username.toLowerCase().replace(/[|\s&;$%&§\{\[\]\}@"<>()+,!?\-._~\=\*\+\#\;\\/]/g, ''), this);
+            emit(this.username.toLowerCase().replace(/[|\s&;$%&§{[\]}@"<>()+,!?\-._~=*+#;\\/]/g, ''), this);
```
Comment (rmeissn) [inline application/db_migration_usernames.js]: Please use the old regex and deactivate eslint for this line.

File: application/db_migration_usernames.js [hunk 2]
```diff
@@ -55,14 +58,14 @@ return helper.connectToDatabase()
               .then((db) => db.collection('users_duplications'))
               .then((coll) => coll.mapReduce(function() {
                 if (this.value.count === undefined || this.value.count === null || this.value.count < 1) {
-                  this.value.username = this.value.username.replace(/[|\s&;$%&§\{\[\]\}@"<>()+,!?\=\*\+\#\;\\/]/g,'_');
+                  this.value.username = this.value.username.replace(/[|\s&;$%&§{[\]}@"<>()+,!?=*+#;\\/]/g,'_');
```
Comment (rmeissn) [inline application/db_migration_usernames.js]: Please use the old regex and deactivate eslint for this line.

File: application/db_migration_usernames.js [hunk 3]
```diff
@@ -55,14 +58,14 @@ return helper.connectToDatabase()
...
                   emit(this.value._id, this.value);
                 }
                 else {
                   let i = 0;
                   this.value.users.forEach(function(user) {
                     if (user !== undefined) {
-                      let username = user.username.replace(/[|\s&;$%&§\{\[\]\}@"<>()+,!?\=\*\+\#\;\\/]/g,'_');
+                      let username = user.username.replace(/[|\s&;$%&§{[\]}@"<>()+,!?=*+#;\\/]/g,'_');
```
Comment (rmeissn) [inline application/db_migration_usernames.js]: Please use the old regex and deactivate eslint for this line.

File: application/tests/unit_database_helper.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
+// example unit tests
```
Comment (rmeissn) [inline application/tests/unit_database_helper.js]: The entire file might have been added on coincidence. Please recheck

**Review by rmeissn — CHANGES_REQUESTED** [review]
Looks okay so far. Please answer to the comments of the review and merge latest template again, as another PR has been merged into it.

**Comment by TBoonX** [comment]
@rmeissn please review again

**Comment by rmeissn** [comment]
All questions have been answered/problems were resolved. As still all unit/integrations tests pass, I'm fine with merging it.

But we should still wait for the 2.4 release, before merging this into master, right? @bwulff

**Review by rmeissn — APPROVED** [review]

**Comment by rmeissn** [comment]
Please resolve the conflicts and than merge these changes to master.

**Pull request merged**